### DISCUSSION TITLE
Update on PRs, HACS, and updates...

### DISCUSSION THREAD

**Discussion by BJReplay** [BJReplay/ha-solcast-solar#24] [Announcements]
Hi All,

- HA 2024.6 Warnings
- [HACS PR](https://github.com/hacs/default/pull/2535)
- Travel
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 

As many have noted, HA 2024.6 throws warnings on startup.  One set of warnings can easily be addressed, and has been in [PR#21](https://github.com/BJReplay/ha-solcast-solar/pull/21) (file I/O), but another cannot be easily be addressed without breaking the integration in anyone running an older version of HA (e.g. [PR#20](https://github.com/BJReplay/ha-solcast-solar/pull/20) which addresses the loading of timezone data).

The timezone data warning is just that - a warning - and from my observation, just about every integration is now throwing that warning.

The fix - to use the new async method introduced in 2024.6 will break anyone who uses this integration in an earlier version of HA.

If the old oziee repo was still around, that wouldn't be a problem, but given that it isn't, I don't want to introduce a breaking change so soon after another breaking change.

I know I can change the manifest to require 2024.6, but I'm inclined to wait until everyone has had a chance to upgrade.

If anyone knows of a way of knowing which versions of HA are running (does HA phone home?) so we can assess when most users have upgraded, that would be great.

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 

[This PR](https://github.com/hacs/default/pull/2535) will add the integration back into HACS so that it can be added normally through HACS.  It's hard to tell how long it will be until it's reviewed - there are 270 PRs open for adding to HACs - some are as old as August 2023.   **You can probably help it get reviewed quicker by giving it a thumbs up!**
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 

I'll be travelling from June 21 to July 26, so will only have (even more sporadic than usual) ability to monitor this repo.

**Comment by autoSteve** [comment]
> to use the new async method introduced in 2024.6 will break anyone who uses this integration in an earlier version of HA

Not so. I'll create a pull req.

Happy travels.

**Reply by BJReplay** [reply]
Thanks, I will review on my desktop rather than my phone, but looks good at first glance.

In my mind that was how to do it, but as a python noob, I had no idea of actual implementation.

**Comment by jmccrohan** [comment]
@BJReplay See HA usage statistics [here](https://analytics.home-assistant.io/). Usage of old versions tails off very quickly, so I would introduce that breaking change with a versioned depends sooner rather than holding off.

**Reply by BJReplay** [reply]
> See HA usage statistics [here](https://analytics.home-assistant.io/). Usage of old versions tails off very quickly, so I would introduce that breaking change with a versioned depends sooner rather than holding off.

That's incredibly useful.  Without the PR (that I'm about to review and probably approve) that works around the fact that the new async method wasn't introduced until 2024.6, this shows that almost 19% are on versions older than 2024, but at the same time 35% are already on 2024.6 (which makes sense since three quarters of installs are on OS).

![image](https://github.com/BJReplay/ha-solcast-solar/assets/37993507/63523f98-1b61-4e91-8554-ca03086f43ab)
